cpp/src/arrow/flight/client.cc

```diff
@@ -371,7 +375,7 @@ class GrpcClientInterceptorAdapterFactory
   }
 
  private:
-  std::vector<std::shared_ptr<ClientMiddlewareFactory>> middleware_;
+  std::vector<std::shared_ptr<ClientMiddlewareFactory>>& middleware_;
```

Comment:
Intention is to keep the reference and not a copy?

Reply:
This should be a pointer if not a copy.

Reply:
It was to keep the reference, but you got me thinking more and I think I should just take a copy and then make a function to allow adding to the middleware on the fly, since the issue is that I need to be able to add and remove middleware from the interceptor. I will rejig this.